lib/spack/docs/packaging_guide.rst

```diff
@@ -4280,3 +4289,8 @@ Autotools-based packages would be easy (and should be done by a
 developer who actively uses Autotools).  Packages that use
 non-standard build systems can gain ``setup`` functionality by
 subclassing ``StagedPackage`` directly.
+
+.. Emacs local variables
```

Comment:
Done. Does the commit history match what you want, now?

Reply:
Sorry no I was not clear, I think this should be collapsed into 2 commits: one which contains the `emacs` directives and one that contains all the other changes. The commit messages would have to be "clean" as well (since I plan to merge the commits as-is into develop), e.g. avoid messages like "address review comments from X".

Reply:
@scheibelp How about now?

Reply:
Yes this organization is excellent

BUT I think the first commit message has at least one stale note:

> New git-specific version keyword options `full_depth` and `all_branches` (both Boolean). The former disables the `--depth 1` optimization and the latter the `--single-branch` optimization.

I think as of now there is only `get_full_repo`

Once that is addressed, this will be set

Reply:
@scheibelp Done, I think. Let me know if there's anything else.